### PR TITLE
Azure resource health connector

### PR DESCRIPTION
**Pull request by cs-omkar-kumbhar**
Description:

PMDB Link: https://pmdb.fortinet.com/ProjectManagement/editprojects.php?id=23597
 
UTCs:
Connector installation verified.
Connector logo verified.
Actions and Playbook list verified.
Playbook tags verified.
All Playbook are in info mode verified.
All Playbook are in inactive mode verified.

 
Verified all actions.

### CODE REVIEW

File: azure-resource-health/info.json [hunk 1]
```diff
@@ -0,0 +1,720 @@
...
+      },
+      {
+        "title": "Auth Code",
+        "required": true,
+        "editable": true,
+        "visible": true,
+        "type": "password",
+        "name": "code",
+        "tooltip": "The authorization code that you acquired in the authorization step.",
+        "description": ""
```
Comment (cs-prerna-joshi) [inline azure-resource-health/info.json]: add description.

File: azure-resource-health/info.json [hunk 2]
```diff
@@ -0,0 +1,720 @@
...
+          "description": "The name of the resource"
+        },
+        {
+          "title": "Filter",
+          "required": false,
+          "editable": true,
+          "visible": true,
+          "type": "text",
+          "name": "filter",
+          "value": "",
```
Comment (cs-prerna-joshi) [inline azure-resource-health/info.json:168]: add placeholder with example.

File: azure-resource-health/operations.py [hunk 3]
```diff
@@ -0,0 +1,165 @@
...
+    try:
+        ms = MicrosoftAuth(config)
+        endpoint = ms.host + endpoint
+
+        token = ms.validate_token(config, connector_info)
+        headers["Authorization"] = token
+        headers["Content-Type"] = "application/json"
+
+        try:
+            logger.error(f"\n-------------- method: {method} || endpoint: {endpoint} || headers: {headers} || params: {params} || data: {data} || verify: {ms.verify_ssl} ---------------\n")
```
Comment (cs-prerna-joshi) [inline azure-resource-health/operations.py]: remove the error log.

File: azure-resource-health/operations.py [hunk 4]
```diff
@@ -0,0 +1,165 @@
...
+        endpoint = f"/subscriptions/{subscription_id}/resourceGroups/{resource_group_name}/providers/{resource_provider_name}/{resource_type}/{resource_name}/providers/Microsoft.ResourceHealth/events"
+    elif function_name == FuncName.GET_EVENT_LIST_FOR_SUBSCRIPTION_ID:
+        endpoint = f"/subscriptions/{subscription_id}/providers/Microsoft.ResourceHealth/events"
+    elif function_name == FuncName.GET_EVENT_LIST_FOR_TENANT_ID:
+        endpoint = "/providers/Microsoft.ResourceHealth/events"
+    return endpoint, query_params
+
+
+def get_availability_status(config, params, connector_info):
+    try:
```
Comment (cs-prerna-joshi) [inline azure-resource-health/operations.py]: remove try except block.

File: azure-resource-health/info.json (hunk 1 above)
Comment (cs-omkar-kumbhar) [reply azure-resource-health/info.json]: done

File: azure-resource-health/info.json (hunk 2 above)
Comment (cs-omkar-kumbhar) [reply azure-resource-health/info.json:168]: done

File: azure-resource-health/operations.py (hunk 3 above)
Comment (cs-omkar-kumbhar) [reply azure-resource-health/operations.py]: done

File: azure-resource-health/operations.py (hunk 4 above)
Comment (cs-omkar-kumbhar) [reply azure-resource-health/operations.py]: done

**Review by cs-prerna-joshi — APPROVED** [review]

**Pull request merged**